ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Dual.hs

```diff
@@ -427,6 +428,21 @@ instance Bridge m a => TableStuff (LedgerState (DualBlock m a)) where
        foldLedgerTables f main
     <> foldLedgerTables f aux
 
+  traverseLedgerTables f (DualBlockLedgerTables main aux) =
+    liftA2 DualBlockLedgerTables
+      (traverseLedgerTables f main)
+      (traverseLedgerTables f aux)
+
+  zip2ALedgerTables
+    f
+    (DualBlockLedgerTables mainL auxL)
+    (DualBlockLedgerTables mainR auxR) =
+      liftA2 DualBlockLedgerTables
+        (zip2ALedgerTables f mainL mainR)
+        (zip2ALedgerTables f auxL  auxR)
+
+  namesLedgerTables = error "DualBlock:namesLedgerTable:not sure how to combine these"
```

Comment:
I think:

```
zipLedgerTables (\(ApplyNameMK l) (ApplyNameMK r) -> ApplyNameMK $ "Dual-" <> l <> "-" <> r) (namesLedgerTables @m) (namesLedgerTables @a)
```

Reply:
Unfortunately I don't think this quite works. the two namesLedgerTables are different data types, they may well have different numbers and types of tables.

Reply:
I'm going with `namesLedgerTables = DualBlockLedgerTables namesLedgerTables namesLedgerTables`  for now

Reply:
> Unfortunately I don't think this quite works. the two namesLedgerTables are different data types, they may well have different numbers and types of tables.

Hmm, that's true in general, yeah. However, I think for our intended uses case (there's a [simulation](https://en.wikipedia.org/wiki/Simulation_(computer_science)) between the two blocks), it's fair -- especially for now -- to assume that they have the same tables.

That said, I think what you went with already matches the existing `LedgerTables (LedgerState (DualBlock m a)) mk` instance perfectly.